### PR TITLE
Switch file operations to use SDL

### LINKED ISSUE #221: X16Emu doesn't run for Windows users with nonstandard names
In a fit of stupidity, I once decided my Windows username should be Björn. On running the exe, I get "Cannot open C:\Users\Bj├Ârn\Desktop\x16emu_win-r34\rom.bin!" and then back to prompt.

### PR DESCRIPTION
**Pull request by Serentty**
This is intended to fix #221, but it has the effect of fixing a few other issues as well, including inconsistency over the encoding of BASIC files. Before merging, it will be necessary to test that all functionality is still working, which is something I'm already working on.

Issue #221 seems fixed by this, as now it is possible to load the ROM from a path containing arbitrary characters.

### CODE REVIEW

File: loadsave.c [hunk 1]
```diff
@@ -132,15 +133,17 @@ LOAD()
...
 			RAM[STATUS] = a;
 			status |= 1;
 			return;
 		}
-		uint8_t start_lo = fgetc(f);
-		uint8_t start_hi = fgetc(f);
+		uint8_t start_lo;
+		SDL_RWread(f, &start_lo, sizeof(start_lo), 1);
+		uint8_t start_hi;
+        SDL_RWread(f, &start_hi, sizeof(start_hi), 1);
```
Comment (greg-king5) [inline loadsave.c]: I'm curious to know why you don't use `SDL_ReadU8()` in place of `fgetc()`, and `SDL_WriteU8()` in place of `fputc()`.

File: vera_uart.c [hunk 2]
```diff
@@ -14,15 +15,28 @@ static uint8_t byte_in;
...
+SDL_RWops *uart_out_file = NULL;
 
 static bool
 txbusy()
 {
 	return countdown_out > 0;
 }
 
+bool
+check_eof(SDL_RWops *f)
```
Comment (greg-king5) [inline vera_uart.c]: That function should be defined as `static`.

File: loadsave.c (hunk 1 above)
Comment (greg-king5) [reply loadsave.c]: In fact, at this place in the code, you can use
```c
uint16_t start = SDL_ReadLE16(f);
```

**Comment by Serentty** [comment]
Oh, so there _is_ a function what does the same thing as ``fgetc()``? I was looking through the SDL documentation and didn't see that because it wasn't listed under the functions for ``SDL_RWops``. I'll fix that and the static thing.

**Comment by greg-king5** [comment]
> Oh, so there _is_ a function what does the same thing as `fgetc()`? I was looking through the SDL documentation, and didn't see that because it wasn't listed under the functions for `SDL_RWops`.

https://wiki.libsdl.org/CategoryIO

**Comment by Serentty** [comment]
I'll get around to making those small fixes soon. I did all of this on a huge binge one night and need to wait until I have that motivation again.

**Comment by mist64** [comment]
Superseded by https://github.com/commanderx16/x16-emulator/pull/256

**Comment by Serentty** [comment]
@mist64 It seems I kept putting this off until it got taken care of anyway. Sorry about that, and thanks for your hard work!

**Pull request closed without merging**